### PR TITLE
chore: bump nfs-ganesha to v6.2

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#9614

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -9,28 +9,29 @@ RUN zypper -n ref && \
...
 
 # build ganesha only supporting nfsv4 and vfs
 # Set NFS_V4_RECOV_ROOT to /tmp we don't support recovery in this release
 # we disable dbus (-DUSE_DBUS=OFF) for the single share manager since we don't use dynamic exports
+ENV CC=/usr/bin/gcc-14 \
+    CXX=/usr/bin/g++-14
 RUN mkdir -p /usr/local \
     && cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_CONFIG=vfs_only \
-    -DUSE_DBUS=OFF -DUSE_NFS3=OFF -DUSE_NLM=OFF -DUSE_RQUOTA=OFF -DUSE_9P=OFF -D_MSPAC_SUPPORT=OFF -DRPCBIND=OFF \
+    -DUSE_DBUS=OFF -DUSE_NLM=OFF -DUSE_RQUOTA=OFF -DUSE_9P=OFF -D_MSPAC_SUPPORT=OFF -DRPCBIND=OFF \
```
Comment (james-munson) [inline package/Dockerfile:34]: Is there a reason that this drops -DUSE_NFS3=OFF ?

**Review by james-munson — COMMENTED** [review]
Looks fine.  One question.

File: package/Dockerfile (hunk 1 above)
Comment (derekbit) [reply package/Dockerfile:34]: nfs-ganesha bug https://github.com/nfs-ganesha/nfs-ganesha/issues/1200#issuecomment-2489968401.
I think enabling the option is not harmful. WDYT?

Comment (james-munson) [reply package/Dockerfile:34]: Not harmful.  I just wondered if there was a specific need, was all.

**Review by james-munson — APPROVED** [review]

**Pull request merged**